extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -0,0 +1,681 @@
+        !$omp target teams distribute parallel do simd collapse(3) &
+        !$omp map(from:optical_props%tau)
+
+        do ibnd = 1, nbnd
+          do ilay = 1, nlay
+            do icol = 1, ncol
+              ! Absorption optical depth  = (1-ssa) * tau = tau - taussa
+              optical_props%tau(icol,ilay,ibnd) = (atau(icol,ilay,ibnd) - ataussa(icol,ilay,ibnd))
+!              optical_props%tau(icol,ilay,ibnd) = (ltau(icol,ilay,ibnd) - ltaussa(icol,ilay,ibnd)) + &
+!                                                  (itau(icol,ilay,ibnd) - itaussa(icol,ilay,ibnd))
+            end do
+          end do
+        end do
+      type is (ty_optical_props_2str)
+        !$acc parallel loop gang vector default(none) collapse(3) &
+        !$acc               copyin(optical_props) copyout(optical_props%tau, optical_props%ssa, optical_props%g)
+        !$omp target teams distribute parallel do simd collapse(3) &
+        !$omp map(from:optical_props%tau, optical_props%ssa, optical_props%g)
+        do ibnd = 1, nbnd
+          do ilay = 1, nlay
+            do icol = 1, ncol
+              tau    = atau   (icol,ilay,ibnd)
+              taussa = ataussa(icol,ilay,ibnd)
+              optical_props%g  (icol,ilay,ibnd) = (ataussag(icol,ilay,ibnd)) / &
+                                                        max(epsilon(tau), taussa)
+              optical_props%ssa(icol,ilay,ibnd) = taussa/max(epsilon(tau), tau)
+              optical_props%tau(icol,ilay,ibnd) = tau
+!              tau    = ltau   (icol,ilay,ibnd) + itau   (icol,ilay,ibnd)
+!              taussa = ltaussa(icol,ilay,ibnd) + itaussa(icol,ilay,ibnd)
+!              optical_props%g  (icol,ilay,ibnd) = (ltaussag(icol,ilay,ibnd) + itaussag(icol,ilay,ibnd)) / &
+!                                                        max(epsilon(tau), taussa)
+!              optical_props%ssa(icol,ilay,ibnd) = taussa/max(epsilon(tau), tau)
+!              optical_props%tau(icol,ilay,ibnd) = tau
+            end do
+          end do
+        end do
+      type is (ty_optical_props_nstr)
+        error_msg = "aerosol optics: n-stream calculations not yet supported"
+      end select
+    end if 
+    !$acc end data
+    !$omp end target data
+  end function aerosol_optics
+  !--------------------------------------------------------------------------------------------------------------------
+  !
+  ! Inquiry functions
+  !
+  !--------------------------------------------------------------------------------------------------------------------
+  function get_aerosol_size_bin(nbin, aero_bin_lo, aero_bin_hi, aero_size) result(ibin)
+    integer,                   intent(in   ) :: nbin
+    real(wp), dimension(nbin), intent(in   ) :: aero_bin_lo, aero_bin_hi
+    real(wp),                  intent(in   ) :: aero_size
+
+    integer  :: i, ibin
+
+    character(len=128)      :: error_msg
+
+    ibin = 0
+
+    error_msg = ''
+    if (aero_size .lt. aero_bin_lo(1) .or. aero_size .gt. aero_bin_hi(nbin)) then
```

Comment:
@mjiacono Please use `any_vals_outside()` from `mo_rte_array_until`

Reply:
Size check changed to use any_vals_outside()